app/src/lib/stores/pull-request-store.ts

```diff
@@ -0,0 +1,71 @@
+import { PullRequestDatabase } from '../databases'
+import { GitHubRepository } from '../../models/github-repository'
+import { Account } from '../../models/account'
+import { API, IAPIPullRequest } from '../api'
+import { fatalError } from '../fatal-error'
+
+/** The store for GitHub Pull Requests. */
+export class PullRequestStore {
+  private db: PullRequestDatabase
+
+  public constructor(db: PullRequestDatabase) {
+    this.db = db
+  }
+
+  public async cachePullRequests(
+    repository: GitHubRepository,
+    account: Account
+  ) {
+    const api = API.fromAccount(account)
+
+    const prs = await api.fetchPullRequests(
+      repository.owner.login,
+      repository.name,
+      'open'
+    )
+
+    this.writePullRequests(prs, repository)
+  }
+
+  public async getPullRequests(repository: GitHubRepository) {
+    const gitHubRepositoryID = repository.dbID
+    if (!gitHubRepositoryID) {
+      fatalError(
+        "Cannot get pull requests for a repository that hasn't been inserted into the database!"
+      )
+
+      return []
+    }
+
+    const pullRequests = await this.db.pullRequests
+      .where('repo_id')
+      .equals(gitHubRepositoryID)
+      .sortBy('number')
+
+    return pullRequests
+  }
+
+  private async writePullRequests(
+    pullRequests: ReadonlyArray<IAPIPullRequest>,
+    repository: GitHubRepository
+  ): Promise<void> {
+    const repoId = repository.dbID
+
+    if (!repoId) {
+      fatalError(
+        "Cannot store pull requests for a repository that hasn't been inserted into the database!"
+      )
+
+      return
+    }
+
+    const table = this.db.pullRequests
+    const apiPRs = pullRequests as Array<IAPIPullRequest>
+    const insertablePRs = apiPRs.map(x => {
+      return { repoId, ...x }
+    })
+
+    table.clear()
```

Comment:
We _probably_ wanna do the clear + bulkAdd in a transaction so that they're atomic.